accounts/abi/bind/base.go

```diff
@@ -251,8 +251,14 @@ func (c *BoundContract) FilterLogs(opts *FilterOpts, name string, query ...[]int
 	if opts == nil {
 		opts = new(FilterOpts)
 	}
-	// Append the event selector to the query parameters and construct the topic set
-	query = append([][]interface{}{{c.abi.Events[name].Id()}}, query...)
+
+	var e abi.Event
+	var exist bool
+	if e, exist = c.abi.Events[name]; !exist {
```

Comment:
Prefer the more succinct:

```suggestion
	e, exists := c.abi.Events[name]
	if !exists {
```

and also add an `s` at the end of `exists`